libs/chonky/src/lib.rs

```diff
@@ -1,26 +1,106 @@
 #![doc = include_str!("../README.md")]
 
-/// Adds two numbers together
-///
-/// # Example
+use error_stack::Report;
+use image::DynamicImage;
+use pdfium_render::prelude::*;
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum ChonkyError {
+    #[error("parsing error in pdf")]
+    ReadPdf,
+    #[error("pdfium error")]
+    Pdfium,
+    #[error("write error to system")]
+    Write,
+    #[error("Issues with CLI input")]
+    Arguments,
+}
+
+/// Function to read the pdf
 ///
-/// ```rust
-/// use chonky::add;
+/// # Errors
+/// Will return `Err` if `filename` does not exist or the user does not have
+/// permission to read it.
```

Comment:
Minor code-style: it's more readable if there is a newline after headings
```suggestion
/// # Errors
///
/// Will return `Err` if `filename` does not exist or the user does not have
/// permission to read it.
```

We consistently have this in the repository.

Reply:
Ok will make note for the future thanks!